src/CSnakes.Runtime/PythonEnvironmentBuilder.cs

```diff
@@ -0,0 +1,125 @@
+﻿using Python.Runtime;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace CSnakes.Runtime;
+
+public partial class PythonEnvironmentBuilder
+{
+    private readonly string versionPath;
+    private readonly string pythonLocation;
+    private readonly string version;
+    private PythonEnvironmentInternal? env;
+    private string? virtualEnvironmentLocation;
+    private string[] extraPaths = [];
+
+    private PythonEnvironmentBuilder(string pythonLocation, string version)
+    {
+        versionPath = MapVersion(version);
+        this.pythonLocation = pythonLocation;
+        this.version = version;
+    }
+
+    internal static string MapVersion(string version, string sep = "")
+    {
+        // split on . then take the first two segments and join them without spaces
+        var versionParts = version.Split('.');
+        return string.Join(sep, versionParts.Take(2));
+    }
+
+    public PythonEnvironmentBuilder WithVirtualEnvironment(string path, bool ensureVirtualEnvironment = true)
+    {
+        if (ensureVirtualEnvironment)
+        {
+            EnsureVirtualEnvironment(path);
+        }
+
+        virtualEnvironmentLocation = path;
+        extraPaths = [.. extraPaths, path, Path.Combine(virtualEnvironmentLocation, "Lib", "site-packages")];
+
+        return this;
+    }
+
+    public IPythonEnvironment Build(string home)
+    {
+        if (PythonEngine.IsInitialized && env is not null)
+        {
+            // Raise exception?
+            return env;
+        }
+
+        if (!Directory.Exists(home))
+        {
+            throw new DirectoryNotFoundException("Python home directory does not exist.");
+        }
+
+        if (File.Exists(Path.Combine(home, "requirements.txt")))
+        {
+            InstallPackagesWithPip(home);
+        }
+
+        env = new PythonEnvironmentInternal(pythonLocation, version, home, this, extraPaths);
+
+        return env;
+    }
+
+    private void InstallPackagesWithPip(string home)
+    {
+        ProcessStartInfo startInfo = new()
+        {
+            WorkingDirectory = home,
+            FileName = "pip",
+            Arguments = "install -r requirements.txt"
+        };
+
+        if (virtualEnvironmentLocation is not null)
+        {
+            string venvScriptPath = Path.Combine(virtualEnvironmentLocation, "Scripts");
+            startInfo.FileName = Path.Combine(venvScriptPath, "pip.exe");

```

Comment:
This is Windows specific

Reply:
ah good point, it shouldn't be `.exe` off Windows. Is the rest of the path more consistent?

Reply:
It's bin/pip for Nix and Scripts\pip.exe for Windows. But see latter comment about SoC

Reply:
Ah right. I'll pull this part out as a different PR.